### ISSUE TITLE
genicons.py lacks main() function and a significant amount of code executed on import

### ISSUE THREAD

**Issue by TimidRobot** [creativecommons/licensebuttons#12]
## Describe the bug
[`scripts/genicons.py`](https://github.com/creativecommons/licensebuttons/blob/master/scripts/genicons.py) lacks `main()` function and a significant amount of code executed on import.

## Expected behavior
1. It should refactored to use Python3 per [genicons.py uses Python 2, which has been sunsetted · Issue #9](https://github.com/creativecommons/licensebuttons/issues/9)
2. It should be refactored to use a `main()` function with additional functions reducing the size of the last stanza (given the current version at the time of this issue, see [Line 131](https://github.com/creativecommons/licensebuttons/blob/93f90aeb0a591ee356a3604d00462afb12f89cf9/scripts/genicons.py#L131) through the end of the file)
3. The `main()` function should be called using the convention:
    ```python
    if __name__ == "__main__":
        try:
            main()
        except SystemExit as e:
            sys.exit(e.code)
        except KeyboardInterrupt:
            print("INFO (130) Halted via KeyboardInterrupt.", file=sys.stderr)
            sys.exit(130)
        except Exception:
            print("ERROR (1) Unhandled exception:", file=sys.stderr)
            print(traceback.print_exc(), file=sys.stderr)
            sys.exit(1)
    ```

[2019-03-02: edited syntax for  flake8 compliance]

**Comment by DeboDevelop** [comment]
Hey, I would like to work on this issue but in #9 somebody will refactor the code for python3. Should I wait for that issue to be solved before working on this issue?

**Comment by kgodey** [comment]
Yes, I recommend waiting.

**Comment by kgodey** [comment]
Marking this as not ready until https://github.com/creativecommons/licensebuttons/issues/9 is done.

**Comment by kgodey** [comment]
This is ready now.

**Comment by pavitra14** [comment]
I would like to work on this issue if It is up for grabs.

**Comment by kgodey** [comment]
@pavitra14 I just assigned you another issue. please finish that first.

**Comment by keshav2212** [comment]
@kgodey Hi, can I work on this ?

**Comment by pavitra14** [comment]
I submitted the PR for the previous issue, if available, I'll take this one up.

**Comment by DeboDevelop** [comment]
Please go ahead and take it. I am not working on it. The Mentors are free to do what they want.

**Comment by sp35** [comment]
Since I had already worked on this while I was working with #9 but had decided to wait till #17, I went it it.

**Comment by kgodey** [comment]
I originally assigned this to @sp35 but actually @keshav2212 commented on this first, so reassigning.

In the future @sp35 @keshav2212 please wait for approval before starting work on issues.

**Comment by sp35** [comment]
@kgodey Sorry, I'll ensure this from now on.

**Comment by keshav2212** [comment]
@kgodey Sorry for that, I'll take care of it in future.
